transaction/_manager.py

```diff
@@ -174,13 +175,15 @@ def run(self, func=None, tries=3):
         doc = func.__doc__
         if name != '_':
             if doc:
-                doc = name + '\n\n' + doc
+                doc = name + u'\n\n' + doc
```

Comment:
On Python 3 `func.__doc__` and `func.__name__` are already unicode, decoded according to the codec of the file, unless they're assigned manually. On Python 2 it's the same, except vice versa. So in the common case on Python 2 this is going to take a trip through ASCII, which may not be correct (it depends on what the coding token of the file was).

So, then this should probably still be a native string (assuming we don't really care about the manually-assigned docstring case). And then the decoding of `doc` could be moved up here, outside the retry loop. Where it should probably be UTF-8 instead of ASCII (maybe with the fallback to latin-1 😄 ).